pkg/proxy/winkernel/hcnutils.go

```diff
@@ -22,14 +22,45 @@ package winkernel
 import (
 	"encoding/json"
 	"fmt"
+	"os"
+	"time"
 
 	"github.com/Microsoft/hcsshim/hcn"
 	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/pkg/proxy"
 
 	"strings"
 )
 
-type HostNetworkService interface {
+type IHCN interface {
+	GetNetworkByName(networkName string) (*hcn.HostComputeNetwork, error)
+	ListEndpointsOfNetwork(networkId string) ([]hcn.HostComputeEndpoint, error)
+	GetEndpointByID(endpointId string) (*hcn.HostComputeEndpoint, error)
+	ListEndpoints() ([]hcn.HostComputeEndpoint, error)
+	GetEndpointByName(endpointName string) (*hcn.HostComputeEndpoint, error)
+	ListLoadBalancers() ([]hcn.HostComputeLoadBalancer, error)
+	GetLoadBalancerByID(loadBalancerId string) (*hcn.HostComputeLoadBalancer, error)
+	CreateEndpoint(endpoint *hcn.HostComputeEndpoint, network *hcn.HostComputeNetwork) (*hcn.HostComputeEndpoint, error)
+	CreateLoadBalancer(loadbalancer *hcn.HostComputeLoadBalancer) (*hcn.HostComputeLoadBalancer, error)
+	CreateRemoteEndpoint(endpoint *hcn.HostComputeEndpoint, network *hcn.HostComputeNetwork) (*hcn.HostComputeEndpoint, error)
+	DeleteLoadBalancer(loadbalancer *hcn.HostComputeLoadBalancer) error
+	DeleteEndpoint(endpoint *hcn.HostComputeEndpoint) error
+}
+
+// The RealHCN structure represents the real hcn, so we can call the real hcn.methods whenever we need them.
+// If you want to use hcnutils methods on the real hcn you should create it this way: hcnutils := NewHCNUtils(&RealHCN{})
+// Otherwise, if you want to use it on the fakeHCN (for testing purposes), use it this way: hcnutils := NewHCNUtils(&hcntesting.FakeHCN{})
```

Comment:
Hm, I think it would be simpler to just have ``NewHCNUtils`` with no arguments, in which you'd create a new object for ``hcninstance`` directly, and maybe have a ``newFakeHCNUtils`` for testing purposes. That would make the function a bit easier to use, make the code clearer, and this comment less necessary.